sites/org/components/sanity/utils.mjs

```diff
@@ -23,10 +24,19 @@ export const sanityLoader = ({ query, language, type, slug, order }) => {
     query += ` | order(${order})`
   }
 
-  return sanityClient.fetch(query)
+  query += filters
+
+  if (cache[query]) return cache[query]
+
+  console.warn('making a sanity request', query)
+  const result = await sanityClient.fetch(query)
+  cache[query] = result
+  return result
 }
 
 export const sanityImage = (image, dataset = 'site-content') => {
```

Comment:
I don't think 1 method to load all Sanity images is all that great.
t's a minor thing, but I'd prefer to name this method based on the dataset it is loading the image from.
We will probably end up with a different way to retrieve site images vs user images, so calling it `sanityImage` hides too much complexity under the hood.

I'd prefer to have `sanitySiteImage` (and `sanityUserImage` in the future) because I suspect the way to load these will be very different, and frontend developers should not just assume these two are equal.

Reply:
added additional named methods below. It's true, I'm not clear on how to load the userImages, so it's a guess